code/shared/game_rpc/vehicle/vehicle_setprops.h

```diff
@@ -7,36 +7,64 @@
+        Framework::Utils::Optional<bool> radioOn;
+        Framework::Utils::Optional<int> radioStationId;
+        Framework::Utils::Optional<glm::vec4> rimColor;
+        Framework::Utils::Optional<float> rust;
+        Framework::Utils::Optional<bool> sirenOn;
+        Framework::Utils::Optional<glm::vec4> tireColor;
+        Framework::Utils::Optional<glm::vec4> windowTint;
 
         void FromParameters(VehicleSetProps props) {
-            this->beaconLightsState = props.beaconLightsState;
-            this->licensePlate      = props.licensePlate;
-            this->locked            = props.locked;
-            this->radioId           = props.radioId;
-            this->radioState        = props.radioState;
-            this->sirenState        = props.sirenState;
+            this->beaconLightsOn = props.beaconLightsOn;
+            this->colorPrimary   = props.colorPrimary;
+            this->colorSecondary = props.colorSecondary;
+            this->dirt           = props.dirt;
+            this->fuel           = props.fuel;
+            this->licensePlate   = props.licensePlate;
+            this->lockState      = props.lockState;
+            this->radioOn        = props.radioOn;
+            this->radioStationId = props.radioStationId;
+            this->rimColor       = props.rimColor;
+            this->rust           = props.rust;
+            this->sirenOn        = props.sirenOn;
+            this->tireColor      = props.tireColor;
+            this->windowTint     = props.windowTint;
         }
 
         void Serialize(SLNet::BitStream *bs, bool write) override {
-            beaconLightsState.Serialize(bs, write);
+            beaconLightsOn.Serialize(bs, write);
+            colorPrimary.Serialize(bs, write);
+            colorSecondary.Serialize(bs, write);
+            dirt.Serialize(bs, write);
+            fuel.Serialize(bs, write);
             licensePlate.Serialize(bs, write);
-            locked.Serialize(bs, write);
-            radioId.Serialize(bs, write);
-            radioState.Serialize(bs, write);
-            sirenState.Serialize(bs, write);
+            lockState.Serialize(bs, write);
+            radioOn.Serialize(bs, write);
+            radioStationId.Serialize(bs, write);
+            rimColor.Serialize(bs, write);
+            rust.Serialize(bs, write);
+            sirenOn.Serialize(bs, write);
+            tireColor.Serialize(bs, write);
+            windowTint.Serialize(bs, write);
         }
 
         bool Valid() const override {
-            if (radioId.HasValue() && (radioId() >= -1 && radioId() < 4 /* E_RADIO_POLICE */))
+            if (licensePlate.HasValue() && licensePlate().GetLength() > 7) {
                 return false;
-            if (licensePlate.HasValue() && licensePlate().GetLength() > 7)
+            }
+
+            if (radioStationId.HasValue() && (radioStationId() >= -1 && radioStationId() < 4 /* E_RADIO_POLICE */)) {
                 return false;
+            }
+
```

Comment:
Please explain

Reply:
This code was already there

Reply:
Didn't you swap the conditions positions?

Reply:
Ah yes (and rename radioId to radioStationId)

Alphabetical order, no change in the logic